plutus-contract/src/Plutus/Contract/StateMachine/OnChain.hs

```diff
@@ -104,8 +106,8 @@ data StateMachineInstance s i = StateMachineInstance {
     typedValidator :: TypedValidator (StateMachine s i)
     }
 
-machineAddress :: StateMachineInstance s i -> Address
-machineAddress = validatorAddress . typedValidator
+machineAddress :: StateMachineInstance s i -> CardanoAddress
+machineAddress = validatorCardanoAddress testnet . typedValidator
```

Comment:
Add a TODO comment here saying that anything written with the StateMachine won't be usable on mainnet.